python/lsst/ip/diffim/diffimTools.py

```diff
@@ -307,7 +307,10 @@ def sourceToFootprintList(candidateInList, templateExposure, scienceExposure, ke
 
     for kernelCandidate in candidateInList:
         if not type(kernelCandidate) == afwTable.SourceRecord:
-            raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
+            try:
+                kernelCandidate = kernelCandidate.getSource()
+            except:
+                raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
```

Comment:
Depending on where the kernel candidates come from they are either a lsst.afw.table.SourceRecord or a lsst.meas.algorithms.algorithmsLib.PsfCandidateF.  

That is a much better idea to check the type and convert the whole list; I'll move this to the calling task, ImagePsfMatchTask.makeCandidateList(), which already has some type checking in it.